### PR TITLE
Add configuration for setting the desired segment size on a per-table basis.

### PR DESCRIPTION
**Pull request by mcvsubbu**

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java [hunk 1]
```diff
@@ -109,8 +109,30 @@
...
+        /*
+         * The desired size of a completed realtime segment.
+         * This config is used only if REALTIME_SEGMENT_FLUSH_SIZE (or REALTIME_SEGMENT_FLUSH_SIZE  + ".llc") is set
+         * to 0. Default value of REALTIME_SEGMENT_FLUSH_SIZE is "200M". Values are parsed using DataSize class.
+         *
+         * The value for this configuration should be chosen based on the amount of memory available on consuming
+         * machines, the number of completed segments that are expected to be resident on the machine and the amount
+         * of memory used by consuming machines. In other words:
+         *
+         *    numPartitionsInMachine * (consumingPartitionMemory + numPartitionsRetained * REALTIME_SEGMENT_FLUSH_SIZE)
```
Comment (npawar) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java]: should this be REALTIME_DESIRED_SEGMENT_SIZE in the formula in the comment?
numPartitionsInMachine * (consumingPartitionMemory + numPartitionsRetained * REALTIME_DESIRED_SEGMENT_SIZE)

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java [hunk 2]
```diff
@@ -109,8 +109,30 @@
...
+         *
+         * must be less than or equal to the total memory available to store pinot data.
+         *
+         * Note that consumingPartitionMemory will vary depending on the rows that are consumed.
+         *
+         * Not included here is any heap memory used (currently inverted index uses heap memory for consuming partitions).
+         */
+        public static final String REALTIME_DESIRED_SEGMENT_SIZE = "realtime.segment.flush.segment.desired.size";
```
Comment (npawar) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java]: realtime.segment.desired.size?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdateManager.java [hunk 3]
```diff
@@ -78,28 +80,9 @@ public void clearFlushThresholdUpdater(TableConfig tableConfig) {
    */
   private int getLLCRealtimeTableFlushSize(TableConfig tableConfig) {
```
Comment (npawar) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdateManager.java]: Looks like this method is not being used anymore, can we remove it?

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java (hunk 1 above)
Comment (mcvsubbu) [reply pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java]: yes, will fix.

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java (hunk 2 above)
Comment (mcvsubbu) [reply pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java]: realtime.segment.flush.desired.size?
I realize I have one too many 'segment' in there. I like to keep the realtime.segment.flush in place with the other related configs that start with the same prefix.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdateManager.java (hunk 3 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdateManager.java]: Yup, removed.

**Pull request merged**